test-fixtures/update-rust-component-version.py

```diff
@@ -0,0 +1,117 @@
+import os, json
+import requests
+import datetime
+from github import Github
+
+
+GITHUB_REPO = "mozilla/rust-components-swift"
+SPM_PACKAGE = "Client.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved"
+FIREFOX_PROJECT = "Client.xcodeproj/project.pbxproj"
+github_access_token = os.getenv("GITHUB_TOKEN")
+
+def get_latest_rust_components_version():
+    g = Github(github_access_token)
+    repo = g.get_repo(GITHUB_REPO)
+
+    latest_tag = repo.get_tags()[0].name
+    latest_commit = str(repo.get_tags()[0].commit)
+    only_commit = re.findall(r'"([^"]*)"', latest_commit)
+
+    return (str(latest_tag), str(only_commit[0]))
+
+def read_rust_components_tag_version():
+    # Read Package file to find the current rust-component version
+    f = open(SPM_PACKAGE)
+    data = json.load(f)
+    pin = data["object"]
+
+    for i in pin["pins"]:
+        if i["package"] == "MozillaRustComponentsSwift":
+            # return the json with the RustComponent info
+            json_new_version = i["state"]
+    f.close()
+    # Return the current version and commit
+    return json_new_version["version"], json_new_version["revision"]
+
+def read_project_min_version():
+    line_number = 0
+    list_of_results = []
+    string_to_search = 'https://github.com/mozilla/rust-components-swift'
+
+    with open(FIREFOX_PROJECT) as f:
+        #if 'https://github.com/mozilla/rust-components-swift' in f.read():
+        line_read = 0
+        for line in f:
+            # For each line, check if line contains the string
+            line_number += 1
+            if string_to_search in line:
+                # If yes, then look for the field we are interested in: minimumVersion
+                for i in range(3):
+                    minimumVersion = next(f, '').strip()
+
+                version_found = minimumVersion.find("=")
+                last_line_position = minimumVersion.find(";")
+                current_tag_version = ''
+                # version format: XX.Y.Z
+                for i in range(version_found+2 , last_line_position):
+                    current_tag_version+= minimumVersion[i]
+                # Return the current rust-component version in project
+                return current_tag_version
+
```

Comment:
This looks super fragile 😨

Reply:
I would like to improve it, but I can't find another way to go throught the file and read it.. I has been working previously on the A-S implementation and now in focus-iOS doing same thing